### PR TITLE
Fixes #19103: Add a Rudder postgres system technique

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19103

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: techniques/system/rudderPostgresql/1.0/main.st [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+bundle common rudder_postgreql {
+    !(sles_12_0|sles_12_1|sles_12_2)::
+      "service_name" string => execresult("${paths.systemctl} --no-ask-password list-unit-files --type service | ${paths.awk} -F'.' '{print $1}' | ${paths.grep} -E \"^postgresql-?[0-9]*$\" | tail -n 1", "useshell");
+    sles_12_0|sles_12_1|sles_12_2::
```
Comment (ncharles) [inline techniques/system/rudderPostgresql/1.0/main.st:21]: i wish we could drop support for server SLES12SP2 and earlier

File: techniques/system/rudderPostgresql/1.0/main.st [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    # default is localhost if the rudder_db is unknown
+    # or if the role rudder-server-root is installed
+      "host" string => "localhost";
+    role_rudder_db_server_defined.!role_rudder_server_root::
+      "host" string => "${rudder_servers_by_role.roles[rudder-db]}";
+}
+
+bundle agent system_rudder_postgresql_configuration {
+  vars:
+    # sles 12, sp1 and sp2 don't have by default a systemd postgresql
```
Comment (ncharles) [inline techniques/system/rudderPostgresql/1.0/main.st:38]: this comment is missplaced

File: techniques/system/rudderPostgresql/1.0/main.st [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+bundle common rudder_postgreql {
+    !(sles_12_0|sles_12_1|sles_12_2)::
```
Comment (ncharles) [inline techniques/system/rudderPostgresql/1.0/main.st:19]: where is the vars: ?

**Review by ncharles — CHANGES_REQUESTED** [review]
there seems to be copy and paste error in bundle common

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by ncharles — APPROVED** [review]

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**